lightning/src/offers/offer.rs

```diff
@@ -236,10 +254,15 @@ macro_rules! offer_derived_metadata_builder_methods { ($secp_context: ty) => {
 				features: OfferFeatures::empty(), absolute_expiry: None, issuer: None, paths: None,
 				supported_quantity: Quantity::One, signing_pubkey: node_id,
 			},
-			metadata_strategy: core::marker::PhantomData,
+			metadata_strategy: DerivedMetadata(OfferId(nonce)),
```

Comment:
Well, it wouldn't necessarily need to be done by `OfferBuilder`. But a user like LDK Node is probably going to store the `Offer` so would want to use the `OfferId` as the key after creating it. So it could also just be a method on `Offer` that computes it on the fly.

But, yes, we'd then re-create the `OfferId` when receiving the `InvoiceRequest`.

Reply:
Updated `OfferId` to use the tagged hash of the merkle root.